src/webview/message/api.ts

```diff
@@ -1,7 +1,15 @@
 import { DiffFile } from "diff2html/lib/types";
 import { AppConfig } from "../../extension/configuration";
+import { ViewedState } from "../../extension/viewed-state";
+
+export interface UpdateWebviewPayload {
+  config: AppConfig;
+  diffFiles: DiffFile[];
+  diffContainer: string;
+  viewedState: ViewedState;
```

Comment:
I don't know: so far, we seem to use `DiffFile` explicitly only because we want to parse in the extension to check that there's a valid diff, and we don't want to parse again in the webview. It's not actually obvious that parsing the diff again in the webview would be slower than stringifying and then parsing again the `DiffFile` objects when sending a message to the webview.

Sadly, `diff2html` doesn't give us a good way of going between the DOM it generates and the `DiffFile` objects. That's why we have to have `extractNewFileNameFromDiffName()` – it reverses the internal function `filenameDiff()` from `diff2html` -> `render-utils.ts`. 

We could try to always look up the right `File` by its `diff.newName`; or build a map on receipt of `files` to look up by the path extracted from the DOM, but it feels like an unnecessary dependence on `diff2html` internals, and unnecessary building of a map that we already need for storage anyway, that's why I've kept `viewedState` separate from `diffFiles`.

Reply:
You're right! I had forgotten what the `DiffFile` object was composed of.
Let's keep the way you did. :)